merge/nested_test.go

```diff
@@ -500,6 +512,124 @@ func TestUpdateNestedType(t *testing.T) {
 				),
 			},
 		},
+		"struct_apply_remove_all": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+						  value: 1
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
+			`,
+			APIVersion: "v1",
+			Managed:    fieldpath.ManagedFields{},
+		},
+		"struct_apply_remove_dangling": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
+				struct:
+			`,
+			APIVersion: "v1",
+			Managed: fieldpath.ManagedFields{
+				"default": fieldpath.NewVersionedSet(
+					_NS(
+						_P("struct"),
+					),
+					"v1",
+					true,
+				),
+			},
+		},
+		"struct_apply_update_remove_all": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+					`,
+					APIVersion: "v1",
+				},
+				Update{
+					Manager: "controller",
+					Object: `
+						struct:
+						  name: a
+						  value: 1
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
```

Comment:
This result is wrong, `default` should no longer have owned any fields after the controller stomped them, so they should not be removed.

Reply:
The controller owns only the things that it changed, we can't know it wants to own name or not.

Reply:
But this is still wrong, because controller should still own value and that should be preserved. Yes, that's what I'm trying to change right now.

Reply:
oh that's a good point, it was an update. Yeah, the result here is that `struct.value` should exist and = 1.